### ISSUE TITLE
Release 0.21.2

### ISSUE THREAD

**Issue by TobiGr** [TeamNewPipe/NewPipe#6052]
# Changelogs
### New
- Added channel details option to long-press menu #5851
- Add functionality to rename Playlist Name from playlist interface #5982
- Add queue time #6023

### Improved
- Allow the user to pause while a video is buffering #5929
- Improve colors in white theme, migrate to Android DayNight Theme #5927, #5995
- Stream item improvements #6004

### Fixed
- Fixed overlapping fonts #5813
- Fixed crash on tapping anywhere on video after long-pressing 'Popup' button #5813
- Update URLs to HTTPS in READMEs #5942
- Fixed some random NullPointerExceptions #5944
- Fixed crash in history fragment when interacting with header controls and there are no videos in history #5971
- Fixed hidden icon to the left of notification actions in settings #5935 
- Disabled media tunneling on Hi3798MV200 and cvt_mt5886_eu_1g fixing video playback on Formuler and Zephir Android TV devices #5969, #6024
- NullPointerException on screen rotation during exit fix #5976
- Fix NullPointerException: Attempt to get length of null array in MainActivity #5999
- Fix display of channel details when it has no videos #5963

### Development
- Replace FrameLayout with FragmentContainerView where applicable #5882
- Clean up unnecessary onCreate in settings fragments #5879
- Update gradle wrapper #5866
- About package internal revamp #5881
- Update to ExoPlayer 2.13.2 #5856 
- Rearrange libraries #5867 
- Rename related streams to related items #5974 
- Migrate rest of AlertDialogs to AppCompat #6001

### New
- Extract metadata for youtube, soundcloud & mediaccc TeamNewPipe/NewPipeExtractor#306

### Improved
- [Bandcamp]  Load more featured pages TeamNewPipe/NewPipeExtractor#589 
- [Bandcamp] Added Radio stream segments TeamNewPipe/NewPipeExtractor#607 

### Fixed
- Update URL to HTTPS in README TeamNewPipe/NewPipeExtractor#591
- [Bandcamp]  Fixed featured cover arts not loading TeamNewPipe/NewPipeExtractor#589

### Development
- Code improvements TeamNewPipe/NewPipeExtractor#578 TeamNewPipe/NewPipeExtractor#594
- Rename getRelatedStreams to getRelatedItems and change return type TeamNewPipe/NewPipeExtractor#597
- Fix a test and ignore rest TeamNewPipe/NewPipeExtractor#598, TeamNewPipe/NewPipeExtractor#601, TeamNewPipe/NewPipeExtractor#602
- Add body field to Page and use it TeamNewPipe/NewPipeExtractor#599

# APK for testing
[NewPipe_v0.21.2_RC2.zip](https://github.com/TeamNewPipe/NewPipe/files/6336155/NewPipe_v0.21.2_RC2.zip)
**_Please report only new bugs in this thread._**

# To Do
- [x] Update translations
- [x] Wrong color of volume indicator in player https://github.com/TeamNewPipe/NewPipe/pull/6053#issuecomment-817668506
- [x] Fix SoundCloud extraction (#6176)

**Corresponding PR**: #6053

**Comment by opusforlife2** [comment]
While testing "pause while buffering", I found out that the Play/Pause button vanishes (until the video resumes playing) if you seek by manually tapping on the Seekbar. But it remains if you use the double tap gesture to skip forward or backward.

This also happens in the current release. Is this intentional or a bug?

**Comment by opusforlife2** [comment]
_Moved @AlexN1997's comment from Release PR to this Release Issue_

---
I found a bug:

https://user-images.githubusercontent.com/76731207/114376381-dddaed00-9b85-11eb-8650-ee629ed77dcd.mp4

How to reproduce:
Play a video and adjust the sound

Device: LG G5 Android 8.0
App version: v0.21.2 rc1

**Comment by opusforlife2** [comment]
When I tap on a thumbnail to begin a stream, the video buffers, and then immediately pauses when it is about to play. I first encountered this in #5459, and thought it was a bug in that PR.

But that PR isn't merged, so it's a bug in this RC.

**Comment by TobiGr** [comment]
I get a black volume off icon on dark theme, but cannot reprpduce the other vokume draeable bug:
![Screenshot_20210412-222438_NewPipe_release_0 21 2](https://user-images.githubusercontent.com/17365767/114457303-2a521700-9bde-11eb-9b1e-5589bcb27b36.png)

**Comment by mhmdanas** [comment]
@TobiGr I think I found the cause. I believe the bug was introduced by #5927. While all the other volume icons moved to `drawable-night` were white, the only one whose black version was the one moved to `drawable-night` was `ic_volume_off.xml`.

**Comment by TiA4f8R** [comment]
I found two bugs:

- When rotating your device in a channel which isn't supported by NewPipe, the app crash with the following error:

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ FR
* __Content Language:__ fr-FR
* __App Language:__ fr_FR
* __Service:__ none
* __Version:__ 0.21.1
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to read from field 'android.widget.TextView org.schabi.newpipe.databinding.FragmentChannelBinding.errorContentNotSupported' on a null object reference
	at org.schabi.newpipe.fragments.list.channel.ChannelFragment.showContentNotSupported(ChannelFragment.java:508)
	at org.schabi.newpipe.fragments.list.channel.ChannelFragment.handleResult(ChannelFragment.java:475)
	at org.schabi.newpipe.fragments.list.channel.ChannelFragment.handleResult(ChannelFragment.java:67)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment.doInitialLoadLogic(BaseListInfoFragment.java:115)
	at org.schabi.newpipe.fragments.BaseStateFragment.onViewCreated(BaseStateFragment.java:42)
	at org.schabi.newpipe.fragments.list.channel.ChannelFragment.onViewCreated(ChannelFragment.java:125)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:332)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1440)
	at android.app.Activity.performStart(Activity.java:8109)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3806)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:235)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:215)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:187)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2386)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:213)
	at android.app.ActivityThread.main(ActivityThread.java:8178)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)

```
</details>
<hr>

The message `This content is not supported by NewPipe [...]` for unsupported channels isn't shown, maybe it's related to the crash above.

**Comment by TobiGr** [comment]
Second candidate: [NewPipe_v0.21.2_RC2.zip](https://github.com/TeamNewPipe/NewPipe/files/6336155/NewPipe_v0.21.2_RC2.zip)
- [Bandcamp]  Extract related items
- [Bandcamp] Radio kiosk; navigate between tracks via chapters
- fixed wrong color of volume indicators
- updated translations

**Comment by SameenAhnaf** [comment]
I found four new bugs.

(1) Channel details is not shown for neither bookmarked nor created playlists
![IMG_20210422_153927](https://user-images.githubusercontent.com/79245641/115696663-b7d2fb00-a384-11eb-91c4-483ba68069b9.jpg)
![IMG_20210422_160109](https://user-images.githubusercontent.com/79245641/115696684-bacdeb80-a384-11eb-8b9c-3757c598011d.jpg)

(2) Buffering icon is not shown on full screen. However, it is shown on main player
![IMG_20210422_160211](https://user-images.githubusercontent.com/79245641/115696791-d0dbac00-a384-11eb-9e39-6077e610cc35.jpg)

(3) If 'Black' theme is selected, all menus should be black.
(4) Icons on always ask share menu should be white for dark or black theme.
![IMG_20210422_154238](https://user-images.githubusercontent.com/79245641/115697332-49db0380-a385-11eb-849a-789dac3c390b.jpg)

**Comment by SameenAhnaf** [comment]
I am not sure how switching to background from full screen (by locking) works. I rarely use this feature. That's why, I can't say if it's a bug of this apk or not.

But whenever I used, I noticed that video pauses if interface is not shown. Video switches to background if interface is shown. However, video always switches to background in this apk by turning off the screen although users should be given the choice to pause the video or not.

**Comment by nadiration** [comment]
This maybe related to the above comment, when you share a link from another app and the NewPipe share menu (open with actions) appears and you click an option it doesn't show touch highlights it freezes for a moment and proceeds with your chosen action

**Comment by TobiGr** [comment]
@shekhar12020 Thank you for testing and reporting!

(1) I cannot reproduce this. Can you give me instructions in how to reproduce the bug or upload a screen cast?.

(2) Reproduced. This was most likely caused by #5929. What I have found so far. The loading circle / progress bar is only shown when starting to load the video. Once it has changed the loading state, the loading progress circle is not shown again. No matter which player you use. E.g. start a long video in portrait mode and you'll see the circle. Jump to a position which has not been loaded yet and no circle is shown.

(3) Yes, but this is the behaviour since the black theme was introduced.

(4) Will be fixed by #6151 

@nadiration I cannot reproduce this.

**Comment by SameenAhnaf** [comment]
@TobiGr Thank you so much for your time and efforts.

(1) As you can see, there are problems only in subscription or playlist feed. Sorry for too fast videos. I had to cover within the file size limit.

https://user-images.githubusercontent.com/79245641/115942116-6d5b9680-a4ca-11eb-9c13-db72f5c28bde.mp4

However, such problems don't persist in channel feed or search results.

https://user-images.githubusercontent.com/79245641/115942203-dc38ef80-a4ca-11eb-9c64-e5a80ca021da.mp4


(3) If I am not wrong, you were working on theme. It would have been nice if you looked on this ♥️.

Btw, my name is not Shekhar. If required, please copy my name while mentioning 😅.

**Comment by mhmdanas** [comment]
@SameenAhnaf see #6155. Please keep the discussion related to this problem there from now on.

**Comment by TobiGr** [comment]
@SameenAhnaf I am sorry for tagging the wrong person. I was a little sleepy last night ;)

(1) I looked for the wrong things in the popup. My fault. This should be fixed, but not in this release as @mhmdanas pointed out already.

(2) is going to be fixed by #6161

**Comment by opusforlife2** [comment]
@TobiGr How will the branches be handled? The dev branch already contains merged PRs for the next version. The main branch is only supposed to have hotfixes merged to it directly.
